File: Algorithms/Search Sort/Algorithms/bubbleSort/BubbleSort.java
public class BubbleSort {
  public static void main(String args[]) {
    int arr[] = { 5, 8, 9, 3, 2, 66, 88, 1, 2 };
    int len = arr.length;
    bubbleSort(arr, len);
    for (int i : arr) {
      System.out.println(i + " ");
    }
  }

  static void bubbleSort(int[] arr, int len) {
    int i, j, temp = 0;
    for (i = len - 1; i > 0; i--) {
      for (j = 0; j < i; j++) {
        if (arr[j] > arr[j + 1]) {
          temp = arr[j];
          arr[j] = arr[j + 1];
          arr[j + 1] = temp;
        }
      }
    }
  }
}

File: Algorithms/Search Sort/Algorithms/bubbleSort/BubbleSort.js
const print = arr => {
  arr.forEach(el => {
    console.log(el);
  })
}

const bubbleSort = (arr, size) => {
  let [i, j, aux] = [0, 0, 0];
  for (i = 0; i < parseInt(arr.length / 2); i++){
      if (arr[i] > arr[i + 1]) {
        aux = arr[i];
        arr[i] = arr[i + 1]
        arr[i + 1] = aux; 
      }
    }
  print(arr);
}


// Memória = 5 + 5(n) => complexidade de espaço
// CPU 4 + (3 * 5(n)) => complexidade de tempo

// identificar a maior complexidade de n, descarta-se operações menos relevantes
// A ideia é avaliar como a complexidade aumenta de acordo com a entrada de dados do algoritmo

(function () {
  let arr = [5, 6, 9, 1, 2];
  const size = arr.length;
  bubbleSort(arr, size);
})()
